README.md

```diff
@@ -29,7 +39,29 @@ gulp.task('example', function() {
 });
 ```
 
-### decompress()
+### Custom compression
+
+Files are processed with maximum compression and output files have the `.brotli` extension.
+
+```javascript
+var gulp   = require('gulp');
+var brotli = require('gulp-brotli');
+
+gulp.task('example', function() {
+  return gulp.src('path/to/input')
+    .pipe(brotli.compress({
+      extension: 'brotli',
+      brotliParams: {
```

Comment:
iltorb is only doing bindings for brotli, and I doubt brotli will ever add an `extension` parameter so it should be fine.

Deleting the property is not an option unless we clone the object, as it could have unexpected side-effects, consider this:

```js
var brotliParams = {...};

gulp.task('example', function() {
  return gulp.src('path/to/input')
    .pipe(brotli.compress(brotliParams))
    .pipe(gulp.dest('path/to/output'));
});
```

The second time the `example` task is executed, when doing continuous builds for example, the `extension` property will be gone.

Leaving the `extension` property on the object is likely fine, but if we want to be truly future-proof we could do this:

```js
var brotliParams = {
  // brotli parameters
  quality: 11,
  // gulp parameters
  gulpParams: {
    extension: 'brotli'
  }
};
```

But it would be extra verbose for the decompression part.

I'd go with a flat object for now, and if we really need to we can do a breaking change and adopt proper splitting like `gulp-gzip` in a future version.

gulp itself does just that and gets away with it: "[gulp supports all options supported by node-glob and glob-stream except ignore and adds the following options](https://github.com/gulpjs/gulp/blob/master/docs/API.md#options)"